oidc-primer.html

```diff
@@ -0,0 +1,1219 @@
+</pre>
+   <p>The example token would decrypt as:</p>
+   <p>Token Header:</p>
+<pre class="language-json highlight"><c- p>{</c->
+    <c- f>"alg"</c-><c- p>:</c-> <c- u>"none"</c->
+<c- p>}</c->
+</pre>
+   <p>Token Body:</p>
+<pre class="language-json highlight"><c- p>{</c->
+    <c- f>"jti"</c-><c- p>:</c-> <c- u>"a78b49ef-71c5-4985-a50c-5ef3aef0fd8f"</c->
+<c- p>}</c->
+</pre>
+   <p>Save the refresh token to a persistant store.</p>
+   <h4 class="no-num heading settled" id="authorization-code-pkce-flow-step-21"><span class="content">21. Sends tokens</span><a class="self-link" href="#authorization-code-pkce-flow-step-21"></a></h4>
+   <p>Once the OP has confirmed that everything checks out and all the tokens are generated, it sends
+a response with the tokens in the body:</p>
+   <p>Response (content-type: application/json)</p>
+<pre class="language-json highlight"><c- p>{</c->
+    <c- f>"access_token"</c-><c- p>:</c-> <c- u>"531683bf-fea8-4bca-8976-2de50e5c9a50"</c-><c- p>,</c->
+    <c- f>"id_token"</c-><c- p>:</c-> <c- u>"eyJhbGciOiJFUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiJodHRwczovL2FsaWNlLmNvb2xwb2QuZXhhbXBsZS9wcm9maWxlL2NhcmQjbWUiLCJhdWQiOiJodHRwczovL2RlY2VudHBob3Rvcy5leGFtcGxlL3dlYmlkI3RoaXMiLCJ3ZWJpZCI6Imh0dHBzOi8vYWxpY2UuY29vbHBvZC5leGFtcGxlL3Byb2ZpbGUvY2FyZCNtZSIsImlzcyI6Imh0dHBzOi8vc2VjdXJlYXV0aC5leGFtcGxlIiwianRpIjoiODQ0YTA5NWMtOWNkYi00N2U1LTk1MTAtMWRiYTk4N2MwYTVmIiwiaWF0IjoxNjAzMzg2NDQ4LCJleHAiOjE2MDMzODcwNDh9.T306vT8dmn9gQIMEdG92AM4WRnrhqWZTfDpovwqZ6Zn0mK9yxj0iOVGqXD4CW8-tzDTitNwEGorAo85atL0Oeg"</c-><c- p>,</c->
+    <c- f>"refresh_token"</c-><c- p>:</c-> <c- u>"eyJhbGciOiJub25lIn0.eyJqdGkiOiJhNzhiNDllZi03MWM1LTQ5ODUtYTUwYy01ZWYzYWVmMGZkOGYifQ."</c->
+<c- p>}</c->
+</pre>
+   <ul>
+    <li data-md>
+     <p><code>"access_token": "531683bf-feea8..."</code>: The OAuth 2.0 access token, which may be used at the
+OP’s <code>userinfo_endpoint</code>.</p>
+    <li data-md>
+     <p><code>"id_token": "eyJhbGciOiJFU..."</code>: The ID token we generated. The client will use this to
+extract information such as the user’s WebId. A client will also exchange the ID token for
+an access token at an authorization server.</p>
+    <li data-md>
+     <p><code>"refresh_token":	"eyJhbGciOiJ..."</code>: The refresh token. The client will use this to fetch a
+new ID token when the current token expires.</p>
+   </ul>
+   <h3 class="heading settled" data-level="4.2" id="request-flow"><span class="secno">4.2. </span><span class="content">Request Flow</span><a class="self-link" href="#request-flow"></a></h3>
+   <p><img src="oidc-primer-request.svg" width="980"></p>
+   <p>In this example, Alice has logged into <code>https://decentphotos.example</code> and has completed the
+authentication steps above. She wants to make a request to Bob’s Storage to get a photo album
+information at <code>https://bob.otherpod.example/private/photo_album.ttl</code>. Bob has previously
+granted access to Alice but has not granted access to anyone else.</p>
```

Comment:
I believe the order of these should be reversed. There ought to be an introductory paragraph before the figure to set the stage, otherwise it is hard to understand what this request flow is. Also, perhaps it should more clearly say that it is *the* request flow, there's and actual app requesting the resource the end user is interested in.
```suggestion
   <p>In this example, Alice has logged into <code>https://decentphotos.example</code> and has completed the
authentication steps above. She wants to make a request to Bob’s Storage to get a photo album
information at <code>https://bob.otherpod.example/private/photo_album.ttl</code>. Bob has previously
granted access to Alice but has not granted access to anyone else.</p>
   <p><img src="oidc-primer-request.svg" width="980"></p>

```